builtins/match.go

```diff
@@ -34,9 +38,65 @@ func matchRecord(a, b *runtime.Record) (bool, error) {
 			}
 		} else {
 			return false, runtime.Errorf("Value mismatch: field %s in second Record not found in first", k)
+		}
+	}
+	return true, nil
+}
 
+func matchRecordOf(val, pat []runtime.Object) (bool, error) {
+	i, back_i := 0, -1
+	j, back_j := 0, -1
+	for i < len(val) && j < len(pat) {
+		valueAtI := val[i]
+		patternAtJ := pat[j]
+		i++
+		j++
+
+		switch patternAtJ {
+		case runtime.AnyOrNone:
+			if j == len(pat) { // Optimize trailing * case
+				return true, nil
+			}
+			back_j = j // Pattern Element after *
+			i--        // Allow zero-length match
+			back_i = i // First Value Element which could be matched with that *
+		default: // Literal character or ?
+			if ok, _ := match(valueAtI, patternAtJ); ok {
+				if j == len(pat) && i == len(val) { // if last element in both lists matched
+					return true, nil
+				}
+				break
+			}
+			if back_j < 0 {
+				return false, runtime.Errorf("Pattern doesn't match, Element number %d mismatch", i-1) /* No Backtracking possible */
+			}
+			if i == len(val) {
+				return false, runtime.Errorf("End of first RecordOf reached, second still continues")
+			}
+
+			// Try again from last *, one character later in str.
+			j = back_j
+			back_i++
+			i = back_i
+		}
+		if j == len(pat) && i != len(val) {
+			if back_j < 0 {
+				return false, runtime.Errorf("Second RecordOf is matched entirely, first isn't")
+			}
+			j = back_j
+			back_i++
+			i = back_i
 		}
 	}
+	// reached if i == len(val) || j == len(pat)
+	if len(val) == i {
+		for ; j < len(pat); j++ {
+			if pat[j] != runtime.AnyOrNone {
+				return false, runtime.Errorf("First RecordOf is entirely matched, second isn't")
+			}
+		}
+		return true, nil
```

Comment:
This line can be removed. Less return and break make the code easier to comprehend
```suggestion
```